paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/FileStoreSourceSplitReader.java

```diff
@@ -86,11 +92,21 @@ public RecordsWithSplitIds<BulkFormat.RecordIterator<RowData>> fetch() throws IO
             return new RecordsWithPausedSplit<>();
         }
 
+        if (wakeup) {
```

Comment:
Do we need to wakeup here?

Reply:
I think we need, and it's meant to reset the wakeup flag. If not, we should reset the `wakeup` at the end of this method. And short-cut at the beginning looks better. WDYT?

Reply:
Maybe this should be in `pool`.

Reply:
And you can change method name to `poll`....

Reply:
I mean not only `poll` should wakeup. I think the wakeup should work on the next `fetch` call not on the next `poll`. So after the `fetch` the wakeup should reset to normal.